modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/ReadWriteTransactionImpl.java

```diff
@@ -126,10 +135,14 @@ protected CompletableFuture<Void> finish(boolean commit) {
                                         id()
                                 );
                             } else {
-                                return CompletableFuture.completedFuture(null);
+                                return completedFuture(null);
                             }
                         }
                 );
+
+        mainFinishFut.handle((res, e) -> finishFut.get().complete(null));
```

Comment:
What kind of exceptions do you expect here? Is it a tmp solution until `// TODO: https://issues.apache.org/jira/browse/IGNITE-17688 Add proper exception handling.` is implemented?

Reply:
It is need to complete future both on success and exception.